File: gin/util/hash.go
package util

import (
	"crypto/hmac"
	"crypto/md5"
	"crypto/sha1"
	"crypto/sha256"
	"encoding/hex"
)

/* 哈希 */
type Hash struct{}

/* Md5 */
func (Hash) Md5(data string) string {
	h := md5.New()
	h.Write([]byte(data))
	return hex.EncodeToString(h.Sum(nil))
}

/* Sha256 */
func (Hash) Sha256(data string) string {
	b := sha256.Sum256([]byte(data))
	return hex.EncodeToString(b[:])
}

/* HmacSha1 */
func (Hash) HmacSha1(data string, key []byte) []byte {
	h := hmac.New(sha1.New, key)
	h.Write([]byte(data))
	return h.Sum(nil)
}

/* HmacSha256 */
func (Hash) HmacSha256(data string, key []byte) []byte {
	h := hmac.New(sha256.New, key)
	h.Write([]byte(data))
	return h.Sum(nil)
}

/* Byte转为16进制 */
func (Hash) HexEncode(bytes []byte) string {
	return hex.EncodeToString(bytes)
}

File: java/src/main/java/webmis/util/Hash.java
package webmis.util;

import java.security.MessageDigest;

import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;

import org.springframework.util.DigestUtils;

/* 哈希 */
public class Hash {

  /* Md5 */
  public static String Md5(String data) {
    return DigestUtils.md5DigestAsHex(data.getBytes());
  }

  /* Sha256 */
  public static String Sha256(String data) {
    try {
      MessageDigest md = MessageDigest.getInstance("SHA-256");
      md.update(data.getBytes("UTF-8"));
      return HexEncode(md.digest());
    } catch (Exception e) {
      return "";
    }
  }

  /* HmacSha1 */
  public static byte[] HmacSha1(String data, byte[] key) {
    try {
      Mac hmac = Mac.getInstance("HmacSHA1");
      SecretKeySpec keySpec = new SecretKeySpec(key, "HmacSHA1");
      hmac.init(keySpec);
      return hmac.doFinal(data.getBytes("UTF-8"));
    } catch (Exception e) {
      return null;
    }
  }

  /* HmacSha256 */
  public static byte[] HmacSha256(String data, byte[] key) {
    try {
      Mac hmac = Mac.getInstance("HmacSHA256");
      SecretKeySpec keySpec = new SecretKeySpec(key, "HmacSHA256");
      hmac.init(keySpec);
      return hmac.doFinal(data.getBytes("UTF-8"));
    } catch (Exception e) {
      return null;
    }
  }

  /* Byte转为String */
  public static String Base64Encode(byte[] data) {
    return java.util.Base64.getEncoder().encodeToString(data);
  }

  /* Byte转为16进制 */
  public static String HexEncode(byte[] bytes) {
    StringBuilder sb = new StringBuilder();
    String temp = null;
    for (byte aByte : bytes) {
      temp = Integer.toHexString(aByte & 0xFF);
      if (temp.length() == 1) {
        sb.append("0");
      }
      sb.append(temp);
    }
    return sb.toString();
  }

  /* String转Byte */
  public static byte[] StrToByte(String str) {
    try {
      byte[] res = str.getBytes("UTF-8");
      return res;
    } catch (Exception e) {
      return null;
    }
  }
  
}
